test/e2e/storage/persistent_volumes-vsphere.go

```diff
@@ -122,17 +122,14 @@ var _ = SIGDescribe("PersistentVolumes:vsphere", func() {
 				framework.ExpectNoError(framework.DeletePersistentVolumeClaim(c, pvc.Name, ns), "AfterEach: failed to delete PVC ", pvc.Name)
 			}
 		}
-	})
-	/*
-		Clean up
-
-		1. Wait and verify volume is detached from the node
-		2. Delete PV
-		3. Delete Volume (vmdk)
-	*/
-	framework.AddCleanupAction(func() {
-		// Cleanup actions will be called even when the tests are skipped and leaves namespace unset.
-		if len(ns) > 0 && len(volumePath) > 0 {
+		/*
+			Clean up
+
+			1. Wait and verify volume is detached from the node
+			2. Delete PV
+			3. Delete Volume (vmdk)
+		*/
+		if len(volumePath) > 0 {
```

Comment:
`volumePath`, `pvConfig` and `pvcConfig` is set once for all test specs in the `BeforeEach` so volume should also be removed at last once all test specs are executed.